src/common/config.cc

```diff
@@ -267,8 +263,7 @@ int md_config_t::parse_config_files_impl(const std::list<std::string> &conf_file
      * If cluster name is not set yet, use the prefix of the
      * basename of configuration file as cluster name.
      */
-    const char *fn = c->c_str();
-    std::string name(basename(fn));
+    std::string name = c->substr(c->find_last_of("/") + 1);
```

Comment:
as `c_str()` returns a "const char*", and per POSIX standard[1], `basename()` expects a "char*", and it might change the string passed in. even the freebsd and glibc kindly offers `basename(const char*)` implementation.

so, yeah, i agree that we do need to change for standard compliance and better portability.

maybe we can have

```c++
auto start = c->rfind('/') + 1;
auto end = c->find(".conf", start);
if (end == name.npos) {
  ...
} else {
  cluster = name.substr(start, end - start);
}
---
[1] http://pubs.opengroup.org/onlinepubs/9699919799/